app/_shared/toc/toc-side.tsx

```diff
@@ -0,0 +1,162 @@
+import { useCallback, useEffect, useState } from 'react'
+import Link from 'next/link'
+import styled from 'styled-components'
+import { type Toc } from '../../lib/get-toc'
+
+const TocContainer = styled.aside`
+  max-width: 240px;
+  min-width: 210px;
+  position: absolute;
+  height: 100%;
+  right: 100%;
+  margin-right: 2.5rem;
+`
+
+const TocWrapper = styled.div`
+  position: sticky;
+  top: 5.5rem;
+`
+
+const TocContent = styled.div`
+  padding-block: 0.5rem;
+  padding-inline: 1rem;
+  border-right: 1px solid rgba(0, 0, 0, 0.2);
+`
+
+const TocList = styled.ul`
+  all: unset;
+  display: flex;
+  flex-direction: column;
+  gap: 8px;
+`
+
+const TocItem = styled.li.withConfig({
+  shouldForwardProp: (prop) => prop !== 'level',
+})<{ level: number }>`
+  font-size: 14px;
+  font-weight: 500;
+  color: #777777;
+  transform: scale(1);
+  padding-left: ${({ level }) => (level > 1 ? `${(level - 1) * 8}px` : '0')};
+
+  &.active {
+    font-weight: 600;
+    color: #007cee;
+    transform: scale(1.05);
+  }
+
+  &:hover {
+    font-weight: 600;
+    color: #222222;
+  }
+
+  transition: all 0.125s ease-in-out 0s;
+`
+
+const Text = styled.span`
+  font-size: 14px;
+  font-weight: 600;
+  color: #222222;
+`
+
+interface TocSideProps {
+  tableOfContents: Toc[]
+}
+
+export const TocSide = ({ tableOfContents }: TocSideProps) => {
+  const headingTops = useHeadingPositions(tableOfContents)
+  const [activeToc, setActiveToc] = useState('')
+
+  useEffect(() => {
+    const onScroll = () => {
+      const scrollTop = getScrollTop()
+      if (!headingTops || headingTops.length === 0) return
```

Comment:
이것도 block문으로 적는 걸 추천해요

```suggestion
      if (!headingTops || headingTops.length === 0) {
        return;
      }
```